client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -0,0 +1,376 @@
+	}
+
+	function goToCheckout() {
+		if ( ! isEnabled( 'signup/design-picker-premium-themes-checkout' ) ) {
+			return null;
+		}
+
+		if ( isEnabled( 'signup/seller-upgrade-modal' ) ) {
+			recordTracksEvent( 'calypso_signup_design_upgrade_modal_checkout_button_click', {
+				theme: selectedDesign?.slug,
+			} );
+		}
+
+		const plan = isEligibleForProPlan && isEnabled( 'plans/pro-plan' ) ? 'pro' : 'premium';
+
+		if ( siteSlugOrId ) {
+			const params = new URLSearchParams();
+			params.append( 'redirect_to', window.location.href.replace( window.location.origin, '' ) );
+
+			// The theme upsell link does not work with siteId and requires a siteSlug.
+			// See https://github.com/Automattic/wp-calypso/pull/64899
+			window.location.href = `/checkout/${ encodeURIComponent(
+				siteSlug || urlToSlug( site?.URL || '' ) || ''
+			) }/${ plan }?${ params.toString() }`;
+		}
+	}
+
+	function recordStepContainerTracksEvent( eventName: string ) {
+		const tracksProps = {
+			step: 'design-step',
+			intent: intent,
+		};
+
+		recordTracksEvent( eventName, tracksProps );
+	}
+
+	const handleBackClick = () => {
+		if ( isPreviewingDesign ) {
+			setSelectedDesign( undefined );
+			setIsPreviewingDesign( false );
+			return;
+		}
+
+		goBack();
+	};
+
+	// Track scroll event to make sure people are scrolling on mobile.
+	useTrackScrollPageFromTop( isMobile && ! isPreviewingDesign, flow || '', STEP_NAME );
+
+	// Make sure people is at the top when entering/leaving preview mode.
+	useEffect( () => {
+		window.scrollTo( { top: 0 } );
+	}, [ isPreviewingDesign ] );
+
+	// Don't render until we've fetched the designs from the backend.
+	if ( ! site || isLoadingDesigns ) {
+		return null;
+	}
+
+	if ( selectedDesign && isPreviewingDesign ) {
+		const designTitle = selectedDesign.title;
+		const shouldUpgrade = selectedDesign.is_premium && ! isPremiumThemeAvailable;
+		const previewUrl = getDesignPreviewUrl( selectedDesign, {
+			language: locale,
+			// If the user fills out the site title with write intent, we show it on the design preview
+			siteTitle: intent === 'write' ? siteTitle : undefined,
+			verticalId: siteVerticalId,
+		} );
+
+		const stepContent = (
+			<>
+				<UpgradeModal
+					slug={ selectedDesign.slug }
+					isOpen={ showUpgradeModal }
+					closeModal={ closeUpgradeModal }
+					checkout={ goToCheckout }
+				/>
+				<WebPreview
+					showPreview
+					showClose={ false }
+					showEdit={ false }
+					externalUrl={ siteSlug }
+					showExternal={ true }
+					previewUrl={ previewUrl }
+					loadingMessage={ translate(
+						'{{strong}}One moment, please…{{/strong}} loading your site.',
+						{
+							components: { strong: <strong /> },
+						}
+					) }
+					toolbarComponent={ PreviewToolbar }
+					siteId={ site?.ID }
+					url={ site?.URL }
+					translate={ translate }
+					recordTracksEvent={ recordTracksEvent }
+				/>
+			</>
+		);
+
+		const pickDesignText =
+			selectedDesign?.design_type === 'vertical'
```

Comment:
I am afraid the usage of `vertical` here will be very confusing in the future as all designs (generated or static) will be "verticalized". Is it still possible to rename it to, say, `generated`? cc: @arthur791004

Reply:
(Also I'm pretty sure I'm late but I'd prefer it to just be `type` rather than `design_type` as it is already on the `Design` interface 😂)

Reply:
this is a good point. I'll try to see how we use this so far and if it is possible to change to `generated`

Reply:
Currently, this is for the metrics ao that we can easily know what kind of the design user selected

Reply:
I think we could change this in another PR? let's at least leave it as this for this PR and then look at it seperately

Reply:
Sure, it makes most sense haha